operator/iamkeycontroller/observe.go

```diff
@@ -49,11 +49,8 @@ func (p *IAMKeyPipeline) Observe(ctx context.Context, mg resource.Managed) (mana
 		).RunWithContext(pctx)
 
 	if result != nil {
-		connDetails, err := toConnectionDetails(pctx.iamExoscaleKey)
-		if err != nil {
-			return managed.ExternalObservation{}, fmt.Errorf("cannot parse connection details: %w", err)
-		}
-		return managed.ExternalObservation{ResourceExists: true, ResourceUpToDate: false, ConnectionDetails: connDetails}, nil
+		log.V(2).Info("pipeline that fetches and checks credentials secret returned", "error", err)
```

Comment:
I think we actually need `result` not `err`

```suggestion
		log.V(2).Info("pipeline that fetches and checks credentials secret returned", "error", result)
```

Reply:
sorry, yes, saw it later. Actually it makes more sense to call it `err`